Fix simple cipher

leahzxxz opened the pull request:
Closes #399

leahzxxz commented:
@matthewmorgan hi! not sure why travis is failing - looks like some of the specs failed. i changed some code on the spec file to correct lint errors - maybe i missed something there?

matthewmorgan commented on exercises/simple-cipher/example.js:
```diff
@@ -1,10 +1,10 @@
 'use strict';
 
-var ALPHABET = 'abcdefghijklmnopqrstuvwxyz';
+const ALPHABET = 'abcdefghijklmnopqrstuvwxyz';
```
As I mentioned in the previous PR, we want to always use `var` on this track.

matthewmorgan commented on exercises/simple-cipher/simple-cipher.spec.js:
```diff
@@ -26,21 +26,15 @@ describe('Random key cipher', function () {
-    expect( function () {
-      new Cipher('ABCDEF');
-    }).toThrow(new Error('Bad key'));
+    expect(new Cipher('ABCDEF')).toThrow(new Error('Bad key'));
```
You should revert this change.  The `toThrow` matcher requires a function as an argument, but your change provides it with a `Cipher` object instead.  

I imagine you were seeing an error for the eslint `no-new` rule.  You can disable for this test block with:
```javascript
  /* eslint-disable no-new */
describe('Incorrect key cipher', function () {
...
});
  /* eslint-enable no-new */
```

matthewmorgan commented on exercises/simple-cipher/example.js:
```diff
@@ -17,14 +17,14 @@ function randomUpTo(n) {
 
-  function addEncodedCharacter(character, index, array) {
+  function addEncodedCharacter(character, index) {
     /* jshint validthis:true */
```
Hmm...I missed this the first time around.  We should remove these jshint directives.  I'm not sure if they were merged inadvertently or are artifacts of an older linting setup, but they don't make sense anymore.

matthewmorgan commented on exercises/simple-cipher/simple-cipher.spec.js:
```diff
@@ -24,21 +24,22 @@ describe('Random key cipher', function () {
+    expect(function () {
       new Cipher('');
     }).toThrow(new Error('Bad key'));
   });
```
Be sure to add the complementary `enable` line for the eslint rule.

matthewmorgan replied:
No problem.

You need to define the _end_ of the `disable` block, otherwise the entire rest of the file will be ignored for that ESlint rule.  EG:


```javascript
  /* eslint-disable no-new */
describe('Incorrect key cipher', function () {
...
});
  /* eslint-enable no-new */
```

Notice that this block begins with `disable` and ends with `enable`, which turns that linting rule back on.

leahzxxz replied:
Thanks for explaining! Just pushed the change. Final issue woohoo!

matthewmorgan commented on exercises/simple-cipher/example.js:
```diff
@@ -17,14 +17,14 @@ function randomUpTo(n) {
 
-  function addDecodedCharacter(character, index, array) {
+  function addDecodedCharacter(character, index) {
     /* jshint validthis:true */
```
Remove please.

matthewmorgan commented:
PS very excited to see that this is the final issue to resolve #399!!

matthewmorgan commented:
Nice work!

One more question: do you feel comfortable squashing your work to a single commit?  If not, I am happy to help with that.

matthewmorgan commented:
I can just squash and merge ;)
I was suggesting that you do it (if you haven't before) to learn how.  It's not hard.

You could do it like this:
```bash
git checkout master
git pull upstream/master  # or however you have named the exercism/javascript remote
git checkout fix/simple-cipher
git rebase master  # this will replay your changes in a single commit onto master
```

Assuming you don't have any merge conflicts, just force-push your branch back up to your fork:

```bash
git push -f origin fix/simple-cipher
```

Care to give it a try?

leahzxxz commented:
@matthewmorgan thanks for the instructions! just rebased and fixed merge conflicts. Doesn't look like my commits are squashed though... Did I do it right 😝

matthewmorgan commented:
One more step-- needed to make sure you were sync'd up with master (that was the hardest part.)

Let's see if we can get this done before anyone merges anything else in ahead of us.

```bash
git reset --soft fd567086481e7b85a5bbbb406b905acab72db6bb # last commit on master before your new ones
git commit -am 'Fix linting errors for simple-cipher'
git push -f origin fix/simple-cipher
```

leahzxxz commented:
@matthewmorgan Done! let me know if you need anything else. Really appreciate all the guidance!

matthewmorgan commented:
No problem!  Nice work, thanks for your contributions.
